src/List.mo

```diff
@@ -406,7 +406,10 @@ module {
     Array.tabulate<A>(length, func (i) {
       let popped = pop<A>(list);
       list := popped.1;
-      Option.unwrap<A>(popped.0);
+      switch (popped.0) {
+        case null { loop { assert false } };
```

Comment:
Why not just P.unreachable() in the branch?

Reply:
Because @rossberg didn't suggest changing this one? :)

Currently, there is a benefit to doing the `assert false` in this "inlined" way, since the trap will be more useful later, if and when it occurs.  Otherwise, without a full backtrace, all traps point to `P.unreachable()`.  We need to either inline these "special" functions at compile time (WDYT?) or perhaps give backtraces of traps, or both?  The first thing seems easier in the short term (inline certain functions that we know will always `assert false`).